libdirections-hybrid/src/main/java/com/mapbox/navigation/route/hybrid/internal/MapboxHybridRouter.kt

```diff
@@ -48,16 +49,18 @@ class MapboxHybridRouter(
      * on that state we use either the off-board or on-board router.
      */
     init {
-        networkStatusJob = jobControl.scope.monitorChannelWithException(networkStatusService.getNetworkStatusChannel(), { networkStatus ->
-            when (networkStatus.isNetworkAvailable) {
-                true -> {
-                    routeDispatchHandler.set(offboardRouterHandler)
-                }
-                false -> {
-                    routeDispatchHandler.set(onboardRouterHandler)
-                }
-            }
-        }, networkStatusService::cleanup)
+        networkStatusJob = jobControl.scope.monitorChannelWithException(
+            networkStatusService.getNetworkStatusChannel(),
+            ::onNetworkStatusChanged,
+            networkStatusService::cleanup
+        )
+    }
+
+    internal suspend fun onNetworkStatusChanged(networkStatus: NetworkStatus) {
```

Comment:
```suggestion
    internal fun onNetworkStatusChanged(networkStatus: NetworkStatus) {
```

> Redundant 'suspend' modifier

Reply:
it's not redundant, it has to be `suspend`, because passed `predicate` is suspend
```
fun <T> CoroutineScope.monitorChannelWithException(
    channel: ReceiveChannel<T>,
    predicate: suspend (T) -> Unit,
    onCancellation: (() -> Unit) = {}
): Job
```

Reply:
Interesting the IDE doesn't catch this one 🤔

Reply:
yep, IDE marks it redundant